lib/checks/tables/td-headers-attr-evaluate.js

```diff
@@ -40,8 +48,8 @@ export default function tdHeadersAttrEvaluate(node) {
         isSelf = headers.indexOf(cell.getAttribute('id').trim()) !== -1;
       }
 
-      // Check if the headers are of cells inside the table
-      notOfTable = headers.some(header => !ids.includes(header));
+      // Check if the headers are of headers inside the table
+      notOfTable = headers.some(header => !tableHeaders.includes(header));
```

Comment:
The way this is done now axe will use the same fail message "The headers attribute is not exclusively used to refer to other cells in the table". That message needs to be updated, it should at least say "header cell". I'm not so sure that's enough though. I think it would be better if this check had a separate remediation message for this scenario. Something like "The header attribute refers a data cell, this must be a header cell so that all screen readers announce it".

To do that I think you need a separate array, something like `tdInTable`, and if it's not empty, use `this.data({ messageKey })` to set which type of message. You can look at `non-empty-alt` for an example of how message keys are used.